src/utils.jl

```diff
@@ -174,6 +174,60 @@ end
 kaiming_normal(dims...; kwargs...) = kaiming_normal(Random.GLOBAL_RNG, dims...; kwargs...)
 kaiming_normal(rng::AbstractRNG; kwargs...) = (dims...; kwargs...) -> kaiming_normal(rng, dims...; kwargs...)
 
+"""
+    orthogonal_init([rng=GLOBAL_RNG], dims...; gain = 1)
+
+Return an `Array` of size `dims` which is a (semi) orthogonal matrix, as described in [1]. 
+
+The input must have at least 2 dimensions. 
+
+# Examples
+```jldoctest; setup = :(using LinearAlgebra)
+julia> W = Flux.orthogonal_init(5, 7);
+
+julia> summary(W)
+"5×7 Array{Float32,2}"
+
+julia> W * W' ≈ I(5)
+true
+
+julia> W2 = Flux.orthogonal_init(7, 5);
+
+julia> W2 * W2' ≈ I(7)
+false
+
+julia> W2' * W2 ≈ I(5)
+true
+```
+
+# References
+[1] Saxe, McClelland, Ganguli. "Exact solutions to the nonlinear dynamics of learning in deep linear neural networks", ICLR 2014, https://arxiv.org/abs/1312.6120
+
+"""
+function orthogonal_init(rng::AbstractRNG, dims...; gain = 1)
+  if length(dims) < 2
+    throw(ArgumentError("Only Arrays with 2 or more dimensions are supported"))
+  end
+  if length(dims) == 2
+    rows = dims[1]
+  else
+    rows = dims[end]
+  end
+  cols = div(prod(dims),rows)
+  mat = rows > cols ? randn(Float32, rows, cols) : randn(Float32, cols, rows)
+
+  Q, R = LinearAlgebra.qr(mat)
+  Q = Array(Q) * sign.(LinearAlgebra.Diagonal(R))
+  if rows < cols
+    Q = transpose(Q)
+  end
+
+  return gain * reshape(Q, dims)
+end
+
+orthogonal_init(dims::Integer...; kwargs...) = orthogonal_init(Random.GLOBAL_RNG, dims...; kwargs...)
+orthogonal_init(rng::AbstractRNG; kwargs...) = (dims::Integer...; kwargs...) -> orthogonal_init(rng, dims...; kwargs...)
```

Comment:
This a bug fix we merged recently that you'll need to adopt.

```suggestion
orthogonal_init(rng::AbstractRNG; init_kwargs...) = (dims::Integer...; kwargs...) -> orthogonal_init(rng, dims...; init_kwargs..., kwargs...)
```

Reply:
So that will have to change for both the positions where there is `kwargs` right?

Reply:
No only for the one I listed, since it returns a closure. Previously the arguments to the anonymous function were overwriting the "initial" arguments when constructing the closure.

Reply:
Tests are failing. Could you figure out what is going wrong? Or was it `orthogonal_init(rng::AbstractRNG; init_kwargs..., kwargs...)` ?

Reply:
Can you rebase? Probably you just need the updated tests from that PR.